vowpalwabbit/cbify.cc

```diff
@@ -139,11 +118,13 @@ void predict_or_learn(cbify& data, base_learner& base, example& ec)
   base.predict(ec);
   //data.probs = ec.pred.scalars;
 
-  uint32_t action = data.mwt_explorer->Choose_Action(*data.generic_explorer, StringUtils::to_string(data.example_counter++), ec);
+  vector<float> pdf = a_s_get_scores(ec.pred.a_s);
+  sample s = sample_from_pdf(data.app_seed + data.example_counter++, &pdf[0], (uint32_t)pdf.size());
 
   CB::cb_class cl;
-  cl.action = action;
-  cl.probability = ec.pred.a_s[action-1].score;
+  cl.action = s.index + 1;
+  // TODO: if pdf is not normalized, this prob is off. probably better to use s.probability
+  cl.probability = ec.pred.a_s[s.index].score;
```

Comment:
Yes, we should use the normalized version of probability.

Reply:
done